crates/nargo_cli/src/cli/execute_cmd.rs

```diff
@@ -91,42 +90,64 @@ fn execute_package<B: Backend>(
         nargo::NargoError::SolvingError(err) => err,
         _ => return None,
     };
 
     match solving_err {
-        acvm::pwg::OpcodeResolutionError::UnsatisfiedConstrain {
+        acvm::pwg::OpcodeResolutionError::IndexOutOfBounds {
+            opcode_location: error_location,
+            ..
+        }
+        | acvm::pwg::OpcodeResolutionError::UnsatisfiedConstrain {
             opcode_location: error_location,
         } => match error_location {
             ErrorLocation::Unresolved => {
                 unreachable!("Cannot resolve index for unsatisfied constraint")
             }
-            ErrorLocation::Resolved(opcode_location) => Some(*opcode_location),
+            ErrorLocation::Resolved(opcode_location) => Some((*opcode_location, solving_err)),
         },
         _ => None,
     }
 }
 
-fn report_unsatisfied_constraint_error(
-    opcode_location: Option<OpcodeLocation>,
+fn report_opcode_error(
+    opcode_err_info: Option<(OpcodeLocation, &acvm::pwg::OpcodeResolutionError)>,
     debug: &DebugInfo,
     context: &Context,
 ) {
-    if let Some(opcode_location) = opcode_location {
+    if let Some((opcode_location, opcode_err)) = opcode_err_info {
         if let Some(locations) = debug.opcode_location(&opcode_location) {
             // The location of the error itself will be the location at the top
             // of the call stack (the last item in the Vec).
             if let Some(location) = locations.last() {
-                let message = "Failed constraint".into();
-                CustomDiagnostic::simple_error(message, String::new(), location.span)
-                    .in_file(location.file)
-                    .with_call_stack(locations)
-                    .report(&context.file_manager, false);
+                match opcode_err {
+                    acvm::pwg::OpcodeResolutionError::IndexOutOfBounds {
+                        index,
+                        array_size,
+                        ..
+                    } => {
+                        let message = format!(
+                            "Index out of bounds, array has size {array_size:?}, but index was {index:?}"
+                        );
+                        CustomDiagnostic::simple_error(message, String::new(), location.span)
+                            .in_file(location.file)
+                            .with_call_stack(locations)
+                            .report(&context.file_manager, false);
+                    }
+                    acvm::pwg::OpcodeResolutionError::UnsatisfiedConstrain { .. } => {
+                        let message = "Failed constraint".into();
+                        CustomDiagnostic::simple_error(message, String::new(), location.span)
+                            .in_file(location.file)
+                            .with_call_stack(locations)
+                            .report(&context.file_manager, false);
+                    }
+                    _ => (),
+                }
```

Comment:
These two cases differ only by the message they return, so perhaps we should just return the message from each case then have a single error/report below.

I do see that we are currently ignoring all error types other than these two, although it's not clear to me why. Can we either handle these cases as well or get a comment explaining why we don't handle them here? If we don't want to handle them, it can be an early return so that we can still simplify the match.

Reply:
These are the only two errors that have an `ErrorLocation` associated with them that need to be resolved using the `DebugInfo` struct. The rest of the errors are still reported, but they just use the base message provided from the acvm's `OpcodeResolutionError` enum. 

I will add a comment making this distinction as well as a cleaned up match

Reply:
done